Prevent pessimizing-move build warning

With most versions of clang:

```
../drafter/src/RefractDataStructure.cc:748:53: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
            auto samplesInfo = Merge<EnumElement>()(std::move(CloneElementInfoContainer<T>(data.samples)));
                                                    ^
../drafter/src/RefractDataStructure.cc:748:53: note: remove std::move call here
            auto samplesInfo = Merge<EnumElement>()(std::move(CloneElementInfoContainer<T>(data.samples)));
                                                    ^~~~~~~~~~                                          ~
../drafter/src/RefractDataStructure.cc:749:53: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
            auto defaultInfo = Merge<EnumElement>()(std::move(CloneElementInfoContainer<T>(data.defaults)));
                                                    ^
../drafter/src/RefractDataStructure.cc:749:53: note: remove std::move call here
            auto defaultInfo = Merge<EnumElement>()(std::move(CloneElementInfoContainer<T>(data.defaults)));
                                                    ^~~~~~~~~~                                           ~
2 warnings generated.
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -21,7 +21,7 @@ aliases:
   - &test-clang-base
     <<: *test-base
     environment:
-      CXXFLAGS: '-Werror -Wno-error=unused-function -Wno-error=pessimizing-move -Wno-error=missing-braces'
+      CXXFLAGS: '-Werror -Wno-error=unused-function -Wno-error=missing-braces'
 
 workflows:
   version: 2

--- a/src/RefractDataStructure.cc
+++ b/src/RefractDataStructure.cc
@@ -788,8 +788,8 @@ namespace
 
             auto valuesInfo = Merge<EnumElement>()(std::move(data.values));
             auto hintsInfo = Merge<EnumElement>()(std::move(data.hints));
-            auto samplesInfo = Merge<EnumElement>()(std::move(CloneElementInfoContainer<T>(data.samples)));
-            auto defaultInfo = Merge<EnumElement>()(std::move(CloneElementInfoContainer<T>(data.defaults)));
+            auto samplesInfo = Merge<EnumElement>()(CloneElementInfoContainer<T>(data.samples));
+            auto defaultInfo = Merge<EnumElement>()(CloneElementInfoContainer<T>(data.defaults));
 
             dsd::Array enums;
 
@@ -1055,7 +1055,7 @@ namespace
 
         AttachSourceMap(*key, MakeNodeInfo(property.node->name.literal, sourceMap));
 
-        return std::move(key);
+        return key;
     }
 
     bool validKey(const StringElement& element)

--- a/test/test-ElementInfoUtils.cc
+++ b/test/test-ElementInfoUtils.cc
@@ -141,7 +141,7 @@ SCENARIO("Merging complex ElementInfo", "[ElementInfo][merge]")
         Container c;
         Info i;
         i.value.emplace_back(refract::make_element<refract::StringElement>("test"));
-        i.sourceMap = std::move(make_sourcemap<E>({ { 4, 2 } }));
+        i.sourceMap = make_sourcemap<E>({ { 4, 2 } });
         c.emplace_back(std::move(i));
 
         WHEN("it is merged")
@@ -179,13 +179,13 @@ SCENARIO("Merging complex ElementInfo", "[ElementInfo][merge]")
 
         Info i1;
         i1.value.emplace_back(refract::make_element<refract::StringElement>("test"));
-        i1.sourceMap = std::move(make_sourcemap<E>({ { 4, 2 } }));
+        i1.sourceMap = make_sourcemap<E>({ { 4, 2 } });
         c.emplace_back(std::move(i1));
 
         Info i2;
         i2.value.emplace_back(refract::make_element<refract::NumberElement>(2));
         i2.value.emplace_back(refract::make_element<refract::StringElement>("computadora"));
-        i2.sourceMap = std::move(make_sourcemap<E>({ { 10, 20 } }));
+        i2.sourceMap = make_sourcemap<E>({ { 10, 20 } });
         c.emplace_back(std::move(i2));
 
         WHEN("it is merged")
@@ -273,12 +273,12 @@ SCENARIO("Cloning InfoElementsContainer")
 
         Info i1;
         i1.value.emplace_back(refract::make_element<refract::StringElement>("test"));
-        i1.sourceMap = std::move(make_sourcemap<E>({ { 4, 2 } }));
+        i1.sourceMap = make_sourcemap<E>({ { 4, 2 } });
         c.emplace_back(std::move(i1));
 
         Info i2;
         i2.value.emplace_back(refract::make_element<refract::NumberElement>(2));
-        i2.sourceMap = std::move(make_sourcemap<E>({ { 10, 20 } }));
+        i2.sourceMap = make_sourcemap<E>({ { 10, 20 } });
         c.emplace_back(std::move(i2));
 
         WHEN("It is cloned")

